### DISCUSSION TITLE
Flow registered with schedule does not execute?

### DISCUSSION THREAD

**Discussion by sgasse** [PrefectHQ/prefect#4740] [Q&A] [answered]
Hi!

I am pretty new to Prefect but like the concepts. However I have an issue with scheduled runs and could not solve it with the documentation or Google. Maybe I am misunderstanding something very obvious.

On a VM, I have `prefect server` and a local agent running. I registered the exemplary `say_hello` flow with the schedule from the documentation. In the UI of the server, I can see the flow and also trigger a run manually / schedule a run for a later specific point. Though there are no runs listed in `Upcoming runs` just from the schedule I specified at registration time. You can see in the UI that the flow has a schedule attached and that it is switched on.
- Do I misunderstand something and is another step needed to have a scheduled flow active?
- Are scheduled runs not shown in `Upcoming runs`?

![Screenshot from 2021-07-06 11-27-40](https://user-images.githubusercontent.com/47532708/124578327-677d1c80-de4e-11eb-8dc8-6a4076f35266.png)

Adding a schedule manually does not change anything either. And I don't think it is something like 'no matching agent for the labels' since I can trigger a run manually.

The code is as in the example (plus idempotency key and tags):
```python3
from datetime import timedelta

from prefect import task, Flow
from prefect.schedules import IntervalSchedule


@task
def say_hello():
    print("Hello, world!")


schedule = IntervalSchedule(interval=timedelta(minutes=2))

with Flow("Scheduled Hello", schedule) as flow:
    say_hello()

flow.register(project_name='example',
              labels=['local'],
              idempotency_key=flow.serialized_hash())
```

Any help would be greatly appreciated! :slightly_smiling_face:

**Comment by zanieb** [comment]
Hi @sgasse -- I can't reproduce this with Prefect Cloud and I presume there is something wrong with the scheduler on your Server instance. The scheduler should be running in the `towel` container. Are there any logs there?

**Reply by sgasse** [reply]
Hey @madkinsz , thank you for supporting on this issue! In between, I was also on the public slack and it indeed seems to be an issue which only occurs with a local server, not the cloud version. However another user had the same problem and @kvnkho was able to reproduce this, therefore I do not suspect that it is a problem of my particular setup.

Looking at the output of the running docker-compose deployment, I do find some suspicious lines:
```
towel_1     | {"severity": "ERROR", "name": "prefect-server.api.flows", "message": "{'type': ['Missing da
ta for required field.']}"}
towel_1     | {"severity": "CRITICAL", "name": "prefect-server.api.flows", "message": "Failed to deserial
ize schedule for flow bf4cff19-30a9-4eec-8b80-dff3a8ca6fe6: {'clocks': [{'type': 'IntervalClock', 'labels': None, 'end_date': None, 'interval': 300000000, 'start_date': {'dt': '2021-07-06T09:19:21.276321', 'tz'
: 'UTC'}, '__version__': '0.15.0', 'parameter_defaults': {}}], 'filters': [], 'or_filters': [], '__version__': '0.15.0', 'adjustments': [], 'not_filters': []}"}
towel_1     | {"severity": "ERROR", "name": "prefect-server.api.flows", "message": "{'type': ['Missing data for required field.']}"}
towel_1     | {"severity": "CRITICAL", "name": "prefect-server.api.flows", "message": "Failed to deserialize schedule for flow 71d46bd5-3f33-403a-a983-84ee0e82870f: {'clocks': [{'type': 'IntervalClock', 'labels
': None, 'end_date': None, 'interval': 120000000, 'start_date': None, '__version__': '0.15.0', 'parameter
_defaults': {}}], 'filters': [], 'or_filters': [], '__version__': '0.15.0', 'adjustments': [], 'not_filte
rs': []}"}
towel_1     | {"severity": "INFO", "name": "prefect-server.Scheduler", "message": "Scheduled 0 flow runs.
"}
```
I have two flows scheduled, that is why there are two different intervals.

What could be the required field for which the data is missing so that the deserialization of the flow fails?

**Comment by zanieb** [comment]
Thanks for those logs! This looks related to our 0.15.0 release -- we'll investigate and get back to you.

**Reply by sgasse** [reply]
Great, thank you! :slightly_smiling_face: Please let me know if I can further help in any way.

**Comment by zanieb** [comment, marked as answer]
`prefect server start --version master` should work now. The default 0.15.0 tag will be updated soon.

**Reply by sgasse** [reply]
Amazing, it works now! :smiley: Thank you for the quick help @madkinsz !

About the tag: Do you mean you will release a version `0.15.1` in a few days with the fix? What I did right now was pulling the master images and pinning them via their `sha256`:

```
    image: prefecthq/apollo@sha256:65ca4f0a4a667fd48360a5c00b4670a796dbc43f2e3a1a0e40ad8b40e1e5cd7f
    ...
    image: prefecthq/server@sha256:0490b527756a53bd3c80513cf09d8c51fdf002c7991524898f62fdd43a31a343
```

**Reply by zanieb** [reply]
You're welcome!

The issue here was that our images are dependent on each other so the server:core-0.15.0 actually had core 0.14.x installed in it. We've released new server images (:master, :latest, :2021.07.06) and pushed the :core-0.15.0 tag on the new images as well. Any of those tags has the fix.
